test: use --port=0 in most sequential debugger tests and move them to parallel

joyeecheung opened the pull request:
#### test: use --port=0 in debugger tests that do not have to work on 9229

To avoid failures when there is another running process occupying
the port 9229 which may happen if there is a stale process, use the
--port argument of node-inspect to use a random port in tests that
don't have to work on port 9229.

The following tests are not touched:

- test-debugger-custom-port: tests a specific port
- test-debugger-debug-brk: tests a specific port
- test-debugger-invalid-args: tests other inspect combinations
- test-debugger-pid: node-inspect does not support -p and --port
  together
- test-debugger-launch: tests that default port is 9229


Based on: https://github.com/nodejs/node/pull/44342

#### test: move debugger tests with --port=0 to parallel

Before

```
❯ tools/test.py "sequential/test-debugger*"
[00:25|% 100|+  32|-   0]: Done

All tests passed.
❯ tools/test.py -J "parallel/test-debugger*"
[00:05|% 100|+   6|-   0]: Done

All tests passed.
```

After

```
❯ tools/test.py "sequential/test-debugger*"
[00:06|% 100|+   5|-   0]: Done

All tests passed.
❯ tools/test.py -J "parallel/test-debugger*"
[00:05|% 100|+  33|-   0]: Done

All tests passed.
```

Refs: https://github.com/nodejs/node/issues/47146

<!--
Before submitting a pull request, please read
https://github.com/nodejs/node/blob/HEAD/CONTRIBUTING.md.

Commit message formatting guidelines:
https://github.com/nodejs/node/blob/HEAD/doc/contributing/pull-requests.md#commit-message-guidelines

For code changes:
1. Include tests for any bug fixes or new features.
2. Update documentation if relevant.
3. Ensure that `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

targos reviewed (approved):
LGTM as long as we still have at least one test with the default port.

joyeecheung commented:
> LGTM as long as we still have at least one test with the default port.

Good point, I moved `test-debugger-launch.mjs` back in sequential with the 9229 regex added back and added a comment about why it has to be there.

MoLow reviewed (approved):
Maybe we can run a stress test to make sure this dose not introduce flakyness

joyeecheung commented:
stress test https://ci.nodejs.org/job/node-stress-single-test/388/

joyeecheung commented:
No failures from the stress tests nodes that can actually compile and run the tests (looks like there are several nodes where the job itself is broken though)

joyeecheung commented:
Landed in dfbad862e079...812e236df935
